test/e2e/node/taints.go

```diff
@@ -337,6 +341,37 @@ var _ = SIGDescribe("NoExecuteTaintManager Single Pod [Serial]", func() {
 			framework.Failf("Pod was evicted despite toleration")
 		}
 	})
+
+	// 1. Run a pod with finalizer
+	// 2. Taint the node running this pod with a no-execute taint
+	// 3. See if pod will get evicted and has the pod disruption condition
+	// 4. Remove the finalizer so that the pod can be deleted by GC
+	ginkgo.It("pods evicted from tainted nodes have pod disruption condition", func() {
```

Comment:
if this is tested at integration level already, I don't see much value on this test.

What you can do to add more value is to have a job failure policy that matches the condition. If you agree, I think we can still merge this today and leave the follow up for after code freeze and before test freeze.

Reply:
in doubt add both XD

Reply:
As for the tests mixing job feature policy with pod disruption conditions I prefer to decouple them into a separate PR.

Reply:
that is true... the permission issue by itself is enough justification for the test.

I think you can add a label that matches the feature gates, so that when the features are disabled, the tests don't run.

Debugging shouldn't be a problem if we have good logging. If adding the test motivates us to improve our logging, that's also a win.